ballerina/tests/bulk_csv_insert.bal

```diff
@@ -0,0 +1,289 @@
+    };
+    error|BulkJob insertJob = baseClient->createJob(payload, INGEST);
+
+    if insertJob is BulkJob {
+        string[][] csvContent = check io:fileReadCsv(csvContactsFilePath);
+        foreach int currentRetry in 1 ..< maxIterations + 1 {
+            error? response = baseClient->addBatch(insertJob.id, csvContent);
+            if response is error {
+                test:assertFail(response.message());
+            } else {
+                break;
+            }
+        }
+
+        //get job info
+        foreach int currentRetry in 1 ..< maxIterations + 1 {
+            error|BulkJobInfo jobInfo = baseClient->getJobInfo(insertJob.id, INGEST);
+            if jobInfo is BulkJobInfo {
+                test:assertTrue(jobInfo.id.length() > 0, msg = "Getting job info failed.");
+            } else {
+                if currentRetry != maxIterations {
+                    log:printWarn("getJobInfo Operation Failed! Retrying...");
+                    runtime:sleep(delayInSecs);
+                } else {
+                    log:printWarn("getJobInfo Operation Failed! Giving up after 5 tries.");
+                    test:assertFail(msg = jobInfo.message());
+                }
+            }
+        }
+        runtime:sleep(10);
+        //close job
+        future<BulkJobInfo|error> closedJob = check baseClient->closeJob(insertJob.id);
+        BulkJobInfo|error closedJobInfo = wait closedJob;
+        if closedJobInfo is BulkJobInfo {
+            test:assertTrue(closedJobInfo.state == "JobComplete", msg = "Closing job failed.");
+        } else {
+            test:assertFail(msg = closedJobInfo.message());
+        }
+        string[][] jobstatus = check baseClient->getJobStatus(insertJob.id, "successfulResults");
+        foreach string[] item in jobstatus {
+            batchId += item[0] + "\n";
+        }
+
+    } else {
+        test:assertFail(msg = insertJob.message());
+    }
+}
+
+@test:Config {
+    enable: true
+}
+function insertCsvStringArrayFromFile() returns error? {
+    log:printInfo("baseClient -> insertCsvStringArrayFromFile");
+
+    string csvContactsFilePath = "ballerina/tests/resources/contacts2.csv";
+
+    //create job
+    BulkCreatePayload payload = {
+        'object : "Contact",
+        contentType : "CSV",
+        operation : "insert",
+        lineEnding : "LF"
+    };
+    error|BulkJob insertJob = baseClient->createJob(payload, INGEST);
+
+    if insertJob is BulkJob {
+        io:ReadableByteChannel|io:Error rbc = io:openReadableFile(csvContactsFilePath);
+        if rbc is io:ReadableByteChannel {
+            foreach int currentRetry in 1 ..< maxIterations + 1 {
+                error? response = baseClient->addBatch(insertJob.id, rbc);
+                if response is error {
+                    test:assertFail(response.message());
+                } else {
+                    break;
+                }
+            }
+            // close channel.
+                _ = check rbc.close();
```

Comment:
```suggestion
            // close channel.
            _ = check rbc.close();
```

Reply:
Updated in [update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/311/commits/cd6ae536defa51bd247b84742f1c480c2cf80ab1)